### ISSUE TITLE
Last resort: non-reproducable style.Seal() exception

### ISSUE THREAD

**Issue by AntonVandeghinste** [MahApps/MahApps.Metro#3549]
Ok, so here's the deal.

I've been Googling for two weeks , to no avail. This post is kind of against all regulations, since i see everyone asking for code samples that make the issue reproducible.
Well that is impossible for my case...

In our company we're remodelling our application from winforms to WPF, and to have a neat layout we're using MaterialdesignInXAML toolkit ([https://github.com/MaterialDesignInXAML/]) , and MahApps ([https://mahapps.com/]).

In one solution of our application, a weird thing happens. The solution builds fine. Two weeks ago, it ran perfectly too.
So , nothing changed since then, and i mean literally NOTHING. We use source control, so i actually made sure i had the EXACT same source code as two weeks ago. But, alas, the solution no longer runs...

But only on my computer that is. All my colleagues have tried and could run perfectly, not a single issue...
My stacktrace:

```
Unhandled Exception: System.Windows.Markup.XamlParseException: Initialization of 'MahApps.Metro.Controls.MetroThumbContentControl' threw an exception. ---> System.InvalidOperationException: Can only base on a Style with target type that is base type 'MetroThumbContentControl'.
   at System.Windows.Style.Seal()
   at System.Windows.StyleHelper.SealIfSealable(Object value)
   at System.Windows.ResourceDictionary.SealValue(Object value)
   at System.Windows.ResourceDictionary.OnGettingValue(Object key, Object& value, Boolean& canCache)
   at System.Windows.ResourceDictionary.OnGettingValuePrivate(Object key, Object& value, Boolean& canCache)
   at System.Windows.ResourceDictionary.GetValueWithoutLock(Object key, Boolean& canCache)
   at System.Windows.ResourceDictionary.GetValue(Object key, Boolean& canCache)
   at System.Windows.SystemResources.LookupResourceInDictionary(ResourceDictionary dictionary, Object key, Boolean allowDeferredResourceReference, Boolean mustReturnDeferredResourceReference, Boolean& canCache)
   at System.Windows.SystemResources.FindDictionaryResource(Object key, Type typeKey, ResourceKey resourceKey, Boolean isTraceEnabled, Boolean allowDeferredResourceReference, Boolean mustReturnDeferredResourceReference, Boolean& canCache)
   at System.Windows.SystemResources.FindResourceInternal(Object key, Boolean allowDeferredResourceReference, Boolean mustReturnDeferredResourceReference)
   at System.Windows.StyleHelper.GetThemeStyle(FrameworkElement fe, FrameworkContentElement fce)
   at System.Windows.FrameworkElement.UpdateThemeStyleProperty()
   at System.Windows.FrameworkElement.OnInitialized(EventArgs e)
   at System.Windows.FrameworkElement.TryFireInitialized()
   at MS.Internal.Xaml.Runtime.ClrObjectRuntime.InitializationGuard(XamlType xamlType, Object obj, Boolean begin)
   --- End of inner exception stack trace ---
   at System.Windows.Markup.XamlReader.RewrapException(Exception e, IXamlLineInfo lineInfo, Uri baseUri)
   at System.Windows.FrameworkTemplate.LoadTemplateXaml(XamlReader templateReader, XamlObjectWriter currentWriter)
   at System.Windows.FrameworkTemplate.LoadTemplateXaml(XamlObjectWriter objectWriter)
   at System.Windows.FrameworkTemplate.LoadOptimizedTemplateContent(DependencyObject container, IComponentConnector componentConnector, IStyleConnector styleConnector, List`1 affectedChildren, UncommonField`1 templatedNonFeChildrenField)
   at System.Windows.FrameworkTemplate.LoadContent(DependencyObject container, List`1 affectedChildren)
   at System.Windows.StyleHelper.ApplyTemplateContent(UncommonField`1 dataField, DependencyObject container, FrameworkElementFactory templateRoot, Int32 lastChildIndex, HybridDictionary childIndexFromChildID, FrameworkTemplate frameworkTemplate)
   at System.Windows.FrameworkTemplate.ApplyTemplateContent(UncommonField`1 templateDataField, FrameworkElement container)
   at System.Windows.FrameworkElement.ApplyTemplate()
   at System.Windows.FrameworkElement.MeasureCore(Size availableSize)
   at System.Windows.UIElement.Measure(Size availableSize)
   at System.Windows.Interop.HwndSource.SetLayoutSize()
   at System.Windows.Interop.HwndSource.set_RootVisualInternal(Visual value)
   at System.Windows.Window.SetRootVisual()
   at System.Windows.Window.SetRootVisualAndUpdateSTC()
   at System.Windows.Window.SetupInitialState(Double requestedTop, Double requestedLeft, Double requestedWidth, Double requestedHeight)
   at System.Windows.Window.CreateSourceWindow(Boolean duringShow)
   at System.Windows.Window.ShowHelper(Object booleanBox)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at MS.Internal.CulturePreservingExecutionContext.Run(CulturePreservingExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Windows.Threading.DispatcherOperation.Invoke()
   at System.Windows.Threading.Dispatcher.ProcessQueue()
   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   at System.Windows.Application.RunDispatcher(Object ignore)
   at System.Windows.Application.RunInternal(Window window)
   at <ApplicationName>.<SolutionName>.WPF.TestApp.App.Main()
```

What i have tried (in order):
Make sure windows updates were correctly installed.
Remove any unnecessary extensions/plugins from vs.
Make sure VS versions of my colleages are the same as mine.
Make sure VS installation is identical to my colleages'.
Clear the source control cache , and perform a pull of the code.
Delete the working folder on my computer , clear the source control cache, and pull.
Clear the visual studio cache, reset user preferences/settings, delete working folder, clear source control cache, and pull.
Reinstall Visual Studio.

So basically, i've done almost anything within my knowledge, except the following two:
Reinstall .NET Framework (and libraries and so on , the whole bunch)
Reinstall Windows (Which i would prefer not to).

Also, googling the exact error, gives (for me) one exact match in stacktrace , but with no extra info included.

Please, please help me fix this , because i'm almost getting depressed about this thing.
(slightly modified code files can be supplied, but since my colleages can run perfectly, and the code is "confidential" i will not include them right away.)

Original post: https://social.msdn.microsoft.com/Forums/en-US/caea4832-2f50-47d3-a4f7-0a66499b05bf/nonreproducible-styleseal-issue-last-resort?forum=wpf

Thx!
Anton

**Comment by punker76** [comment]
@AntonVandeghinste But what version of MahApps are you using? Which VS? Which target framework (of your App)?

**Comment by AntonVandeghinste** [comment]
Woops I'm sorry @punker76 :
MahApps version: 1.6.5
VS version: 2017 Professional (v15.9.12)
Target framework: .Net Framework 4.6.2

**Comment by punker76** [comment]
@AntonVandeghinste How/Where have you added the MahApps resources? Can you provide this code? Do you have some extra custom styles or something else?

**Comment by AntonVandeghinste** [comment]
@punker76 
The MahApps resources are referenced through nuget, so pretty standard.
I use the MetroWindow base class for my MainWindow.

`<metro:MetroWindow x:Class="LnxJeuDax.DIAM.UI.WPF.MainWindow"
                   xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
                   xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
                   xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
                   xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
                   xmlns:local="clr-namespace:LnxJeuDax.DIAM.UI.WPF"
                   xmlns:metro="http://metro.mahapps.com/winfx/xaml/controls"
                   xmlns:vm="clr-namespace:LnxJeuDax.DIAM.UI.WPF.ViewModels"
                   xmlns:sys="clr-namespace:System;assembly=System"
                   xmlns:bc="clr-namespace:LnxJeuDax.DIAM.UI.WPF.BaseControls"
                   mc:Ignorable="d"
                   xmlns:materialDesign="http://materialdesigninxaml.net/winfx/xaml/themes"
                   UseNoneWindowStyle="True"
                   TextElement.Foreground="{DynamicResource MaterialDesignBody}"
                   TextElement.FontWeight="Regular"
                   TextElement.FontSize="13"
                   TextOptions.TextFormattingMode="Ideal"
                   TextOptions.TextRenderingMode="Auto"
                   Background="{DynamicResource MaterialDesignPaper}"
                   FontFamily="{DynamicResource MaterialDesignFont}"
                   WindowState="Maximized"
                   ResizeMode="NoResize"
                   FocusManager.FocusedElement="{Binding ElementName=MenuItemsListBox}">
  <Window.Resources>
    <ResourceDictionary>
      <ResourceDictionary.MergedDictionaries>
        <ResourceDictionary Source="./Resources/GlobalResources.xaml"/>
      </ResourceDictionary.MergedDictionaries>
    </ResourceDictionary>
  </Window.Resources>
  <materialDesign:DialogHost>
    <materialDesign:DrawerHost
        IsLeftDrawerOpen="{Binding ElementName=Menu, Path=IsChecked}">
      <materialDesign:DrawerHost.LeftDrawerContent>
       </materialDesign:DrawerHost.LeftDrawerContent>
    </materialDesign:DrawerHost>
  </materialDesign:DialogHost>
</metro:MetroWindow>`

And as I Just noticed , everything renders just fine in the VS Designer, which makes this issue even more weird than it already was...

Apart from MaterialDesign, Mahapps , ControlzEx, Dragablz i have next to no custom styles.
Maybe a few basedon, but these are mostly related to Material...

Please let me know if you need anything else.
Although I don't think it's got anything to do with MahApps, i just feel helpless, and about to give up :)

**Comment by punker76** [comment]
@AntonVandeghinste So you added also this stuff https://github.com/MahApps/MahApps.Metro/wiki/Quick-Start#add-the-resources-in-appxaml-v165-and-older

**Comment by AntonVandeghinste** [comment]
@punker76 
I did not, but have just now, resulting in the exact same exception...

**Comment by timunie** [comment]
@AntonVandeghinste I had a similar exception in the past. It was caused by the `Style BasedOn="Sone invalid Basecontrol here"`. Could you try to comment out your custom styles?

Happy coding
Tim

**Comment by AntonVandeghinste** [comment]
@timunie 
I can, and I will, although i doubt the impact it will have, since it can run perfectly with my colleages...
Gimme a sec, i'll get back to you

**Comment by AntonVandeghinste** [comment]
@timunie 
Alas , still no changes after commenting my own styles.... :(

**Comment by timunie** [comment]
@AntonVandeghinste The last ideas i have is to reinstall all nuget packeges or try the latest prerelease of MahApps. 

Happy coding 
Tim

**Comment by punker76** [comment]
@AntonVandeghinste What contains your `GlobalResources`?

**Comment by AntonVandeghinste** [comment]
[GlobalResources.txt](https://github.com/MahApps/MahApps.Metro/files/3380420/GlobalResources.txt)
@punker76  Sorry it's in TXT Github won't allow anything else.

@timunie i've tried with the latest prerelease, to no avail. I'll get to reinstalling all nugets today

**Comment by timunie** [comment]
@AntonVandeghinste you can just type the following into the Package Manager:

`Update-Package  –reinstall`

see: [https://docs.microsoft.com/de-de/nuget/consume-packages/reinstalling-and-updating-packages#when-to-reinstall-a-package](https://docs.microsoft.com/de-de/nuget/consume-packages/reinstalling-and-updating-packages#when-to-reinstall-a-package)

Happy coding
Tim

**Comment by AntonVandeghinste** [comment]
Hi

I have resolved the issue. Well, I have eliminated the issue that is.
Turns out if I call InitializeComponent() myself in the ctor of my window, the thing crashes, but only on my computer. (and probably a few others too).

I must say, you both have helped significantly by pointing out different options and things i could try, for which i thank you. 

Concerning the issue itself, i've got no clue why this is an issue on my computer and isn't on my colleagues' computers, but remove the call to InitializeComponent() (PresentationFramework will do this itself) and there isn't an issue on any computer.

Thx Guys
@punker76 
@timunie 
![image](https://user-images.githubusercontent.com/13971253/61040645-499c8e80-a3d1-11e9-9ae1-19e94283f72f.png)
